src/main/sensors/compass.c

```diff
@@ -204,6 +205,22 @@ bool compassDetect(magDev_t *magDev, uint8_t *alignment)
 #endif
         FALLTHROUGH;
 
+    case MAG_LIS2MDL:
```

Comment:
It's almost time to use some cleaner way to do initialization.. The code duplication is incredible ...